### DISCUSSION TITLE
Security Onion

### DISCUSSION THREAD

**Discussion by IsaiahRandolph2** [Security-Onion-Solutions/securityonion#8597] [Unsupported Versions]
Ive done a bit of research but can seem to determine the type of install I want for Security Onion in a production environment. We have a main network then a backup network. Switches and firewalls at each of our location 10+. Im assuming distributed.

But after that do I install one manager node, then a forward node for the other backup network or a forward node for each location. Where are forward nodes installed and what is their purpose. Where are search noes installed and what is their purpose.

Is a separate VM required for each node.And is promiscuous mode something that is required for network traffic to be analyzed. Somewhat confused and wanting to crush this project as im an intern at a game development company of my dreams and this would look awesome on my resume!

**Comment by UMHB-InfoSec** [comment]
1) Your question is harder to answer than you may know.

The type of installation you need will depend on the details of how the network you want to monitor is deployed and the amount of traffic you want to inspect.

The simplest deployment would be to find a place in  your network where all the data you want passes through and put a span or tap there. If the traffic volume is low enough you can use a single box.

More complex deployments require more sensors due to too much traffic and/or multiple tap locations being necessary.

2) As to your question about VMs for each node, I would say to avoid VMs completely unless you have no other option.  Adding a hypervisor just adds another layer of complexity.  If you don't have to, don't.

Look up the different SO Node types.  

https://docs.securityonion.net/en/latest/architecture.html

There are a few different types that are designed for different scenarios.  Generally speaking though there are 3 node types in a distributed setup: manager, search, and forward nodes.  Each type of node is on its own server.

As far as I know promiscuous mode IS required to sniff the traffic, this is another reason it is best to avoid virtualizing SO if you can.

3) If I may offer some unsolicited advice, remember to eat the elephant one bite at a time.  Sit down and draw out your network and figure out your tap locations and how many sensors you need.  Then start setting up your environment and progress methodically and logically.  Document everything you do.  If you find yourself just following instructions and don't understand why you are doing the things you are doing, stop.  Keep researching until you DO understand.

Lastly, never forget the golden rule of networking: "Never do anything you can't immediately undo."

Deploying SO can be a daunting task if approached incorrectly.  But by careful planning and execution I am confident you can do it.
